lib/spack/spack/test/cmd/mirror.py

```diff
@@ -45,3 +69,49 @@ def test_mirror_from_env(tmpdir, mock_packages, mock_fetch, config,
         mirror_res = os.listdir(os.path.join(mirror_dir, spec.name))
         expected = ['%s.tar.gz' % spec.format('{name}-{version}')]
         assert mirror_res == expected
+
+
+def test_mirror_crud(tmp_scope):
+    mirror_ = curry_scope_arg(mirror, 1, tmp_scope)
+
+    mirror_('add', 'mirror', 'http://spack.io')
+    output = mirror_('remove', 'mirror')
```

Comment:
While testing, I encountered an issue with `SpackCommand`s where every second call would fail to capture the output of the command.

Prior to this line, I added `import pdb ; pdb.set_trace()` to try and debug the issue.  When I make this change, the output is correctly captured and the tests run without issue.  So far, this is the only way I've been able to successfully run these tests.

These observations lead me to suspect that there is an issue with `SpackCommand`.  This issue appears to present itself in any tests that run such a command more than once.

Reply:
@opadron @scottwittenburg If you add the `capsys` or `capfd` fixtures to your test and put:

```
with capsys.disabled():  # if this doesn't work try capfd.disabled()
    # ... do stuff ...
```

around the problem regions... does that fix things?

Reply:
Wow! That just fixed everything!  Awesome!  What does that fixture do?

Reply:
`capsys` and `capfd` control Pytest IO capture -- at the python `sys.stdout` / `sys.stderr` level and lower at the system file descriptor level.